source/plugin/placeholders/creatingtokens.rst

```diff
@@ -0,0 +1,159 @@
+* An argument string which generally will be provided by a templating engine
+
+This is the minimum as specified by Sponge. 
+
+If your placeholder is unable to provide text because there is insufficient information, the placeholder should return
+an empty ``Text`` and not throw an exception.
+
+.. tip::
+  If you wish to provide the ability to add multiple arguments to your placeholder, consider specifying a way to split 
+  up the argument string.
+  
+  Remember to tell users of your plugin what you expect your argument string to look like.
+
+Example: Default World Name PlaceholderParser
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This ``PlaceholderParser`` attempts to get the default world's name, returning an empty ``Text`` if it cannot be 
+found. It uses the builder to create the parser with ID ``spongedocs:defaultworld``, assuming the plugin has an ID of 
+``spongedocs``.
+
+.. code-block:: java
+    
+    PluginContainer thisPlugin = ...;
+    
+    PlaceholderParser parser = PlaceholderParser.builder()
+        .plugin(this.thisPlugin)
+        .id("defaultworld")
+        .name("Default World Placeholder")
+        .parser(placeholderContext -> {
+            return Sponge.getServer()
+                .getDefaultWorld()
+                .map(x -> x.getWorldName())
+                .orElse(Text.EMPTY);
+        })
+        .build();
+
+Example: Player Location PlaceholderParser
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This ``PlaceholderParser`` attempts to get the player's location in the world. If used without a ``Player`` as the 
+associated object, it returns an empty :javadoc:`Text`. This implements the ``PlaveholderParser`` interface directly.
+
+.. code-block:: java
+
+    public class PlayerLocationPlaceholder implements PlaceholderParser {
+        
+        @Override
+        public String getId() {
+            return "spongedocs:location"
+        }
+
+        @Override
+        public String getName() {
+            return "Location Placeholder"
+        }
+
+        @Override
+        public Text parse(PlaceholderText placeholderText) {
+            placeholderText.getAssociatedReceiver()
+                .filter(x -> x instanceof Player)
+                .map(player -> ((Player) player).getLocation())
+                .map(location -> Text.of("World: ", location.getExtent().getName(), " - ", location.getPosition()))
+                .orElse(Text.EMPTY);
+        }
+    }
+
+
+Example: Current Time PlaceholderParser
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This ``PlaceholderParser`` returns the current time in the server's local timezone. If the string "UTC" is provided as
+the argument string, it returns the current time in the UTC time zone. This implements the ``PlaveholderParser`` 
+interface directly.
+
+.. code-block:: java
+
+    public class PlayerLocationPlaceholder implements PlaceholderParser {
+
+        @Override
+        public Text parse(PlaceholderContext placeholderContext) {
```

Comment:
Please keep the method order the same as the previous example for consistency reasons.

- getId
- getName
- parse